kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java

```diff
@@ -396,32 +404,44 @@ private boolean rotateOnTime(String encodedPartition, Long recordTimestamp, long
     );
 
     log.trace(
-        "Checking rotation on time with recordCount '{}' and encodedPartition '{}'",
+        "Checking rotation on time for topic-partition '{}' "
+            + "with recordCount '{}' and encodedPartition '{}'",
+        tp,
         recordCount,
         encodedPartition
     );
 
     log.trace(
-        "Should apply periodic time-based rotation (rotateIntervalMs: '{}', baseRecordTimestamp: "
+        "Should apply periodic time-based rotation for topic-partition '{}':"
+            + " (rotateIntervalMs: '{}', baseRecordTimestamp: "
             + "'{}', timestamp: '{}', encodedPartition: '{}', currentEncodedPartition: '{}')? {}",
+        tp,
         rotateIntervalMs,
         baseRecordTimestamp,
         recordTimestamp,
         encodedPartition,
         currentEncodedPartition,
         periodicRotation
     );
+    if (periodicRotation) {
+      return true;
+    }
+    return shouldApplyScheduledRotation(now);
```

Comment:
Any reason not to write this as follows?
```suggestion
    return periodicRotation || shouldApplyScheduledRotation(now);
```

Reply:
Good suggestion I fixed it.